File: dart_lang/lib/c_strategy/c_exercise/notification_strategy/strategies/email_notification_strategy.dart
import 'package:dart_lang/c_strategy/c_exercise/notification_strategy/notification_strategy_inteface.dart';
import 'package:dart_lang/c_strategy/c_exercise/user.dart';

class EmailNotificationStrategy implements NotificationStrategy {
  @override
  void sendNotification(User user, String message) {
    print("sending Email... ");
  }
}

File: PHP/Patterns/src/Behavioral/Strategy/ExerciseOne/NotificationStrategys/EmailNotificationStrategy.php
<?php

namespace DesignPatterns\Behavioral\Strategy\ExerciseOne\NotificationStrategys;

use DesignPatterns\Behavioral\Strategy\ExerciseOne\Models\User;
use DesignPatterns\Behavioral\Strategy\ExerciseOne\Interfaces\NotificationInterface;

class EmailNotificationStrategy implements NotificationInterface
{
    /**
     * @param User $user
     * @param String $message
     * @return void
     */
    public function sendNotification(User $user, String $message): void
    {
        echo "Sending email to {$user->getName()} with message: {$message}";
    }
}